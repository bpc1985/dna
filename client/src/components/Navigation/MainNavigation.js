import React from "react";
import { NavLink } from "react-router-dom";
import logo from "./logo.png";
import Logout from "./Logout";
import "./MainNavigation.scss";

export default function MainNavigation({ isAuthenticated, isAuthenticating, userHasAuthenticated }) {
  return (
    !isAuthenticating && (
      <header className="mainMavigation">
        <div className="logo">
          <img src={logo} alt="DNA" />
        </div>
        <nav className="items">
          <ul>
            {!isAuthenticated && (
              <li>
                <NavLink to="/login">Login</NavLink>
              </li>
            )}
            {isAuthenticated && (
              <>
                <li>
                  <NavLink to="/subscriptions">Subscriptions</NavLink>
                </li>
                <li>
                  <NavLink to="/me">Account</NavLink>
                </li>
                <li>
                  <Logout userHasAuthenticated={userHasAuthenticated} />
                </li>
              </>
            )}
          </ul>
        </nav>
      </header>
    )
  );
}
