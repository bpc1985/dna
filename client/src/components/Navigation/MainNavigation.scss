.mainMavigation {
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 3.5rem;
  background: #ff007c;
  padding: 0 2rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  box-sizing: border-box;

  .logo {
    img {
      height: 3.5rem;
    }
  }

  .items {
    margin-left: 1.5rem;

    ul {
      display: flex;
      list-style: none;
      padding: 0;
      margin: 0;
      align-items: center;
    }

    li {
      margin: 0 1rem;
      span {
        color: #dfcefc;
        margin-right: 40px;
      }
    }

    @media (max-width: 420px) {
      li {
        margin: 0 0.2rem;
      }
    }

    a, button {
      text-decoration: none;
      color: white;
      padding: 0.25rem 0.5rem;
      border: none;
      font: inherit;
      background: transparent;
      cursor: pointer;
      vertical-align: middle;
      margin: 0;
    }

    a:hover,
    a:active,
    a.active,
    button:hover,
    button:active {
      background: #ffffff;
      color: #ff007c;
      border-radius: 5px;
    }
  }
}

@media (max-width: 420px) {
  .mainMavigation {
    padding-right: 0.5rem;
  }
}