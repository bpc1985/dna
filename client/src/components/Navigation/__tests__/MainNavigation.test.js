import { render } from "@testing-library/react";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import MainNavigation from "../MainNavigation";

describe("MainNavigation", () => {
  it("should render component when user is logged in", () => {
    const props = {
      isAuthenticated: true,
      userHasAuthenticated: jest.fn()
    };
    const utils = render(
      <BrowserRouter>
        <MainNavigation {...props} />
      </BrowserRouter>
    );
    expect(utils.container).toMatchSnapshot();
  });

  it("should render component when user is not logged in", () => {
    const props = {
      isAuthenticated: false,
      userHasAuthenticated: jest.fn()
    };
    const utils = render(
      <BrowserRouter>
        <MainNavigation {...props} />
      </BrowserRouter>
    );
    expect(utils.container).toMatchSnapshot();
  });
});
