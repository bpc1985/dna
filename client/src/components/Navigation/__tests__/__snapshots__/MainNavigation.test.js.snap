// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNavigation should render component when user is logged in 1`] = `
<div>
  <header
    class="mainMavigation"
  >
    <div
      class="logo"
    >
      <img
        alt="DNA"
        src=""
      />
    </div>
    <nav
      class="items"
    >
      <ul>
        <li>
          <a
            href="/subscriptions"
          >
            Subscriptions
          </a>
        </li>
        <li>
          <a
            href="/me"
          >
            Account
          </a>
        </li>
        <li>
          <button>
            Logout
          </button>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;

exports[`MainNavigation should render component when user is not logged in 1`] = `
<div>
  <header
    class="mainMavigation"
  >
    <div
      class="logo"
    >
      <img
        alt="DNA"
        src=""
      />
    </div>
    <nav
      class="items"
    >
      <ul>
        <li>
          <a
            href="/login"
          >
            Login
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
